### ISSUE TITLE
💅 lint/correctness/noUnreachable false positive

### ISSUE THREAD

**Issue by jer-sen** [biomejs/biome#1827]
### Environment information

```bash
CLI:
  Version:                      1.4.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"      
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/1.22.21"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnreachable

### Playground link

https://biomejs.dev/playground/?code=dAByAHkAIAB7AAoAIAAgAHUAbgBkAGUAZgBpAG4AZQBkADsACgAgACAAdAByAHkAIAB7AAoAIAAgACAAIAB3AGgAaQBsAGUAIAAoAHQAcgB1AGUAKQAgAHsACgAgACAAIAAgACAAIABpAGYAIAAoAEQAYQB0AGUALgBuAG8AdwAoACkAIAA%2BACAAMAApACAAewAKACAAIAAgACAAIAAgACAAIAB1AG4AZABlAGYAaQBuAGUAZAA7AAoAIAAgACAAIAAgACAAIAAgAGIAcgBlAGEAawA7AAoAIAAgACAAIAAgACAAfQAKACAAIAAgACAAfQAKACAAIAB9ACAAZgBpAG4AYQBsAGwAeQAgAHsACgAgACAAIAAgAGkAZgAgACgARABhAHQAZQAuAG4AbwB3ACgAKQAgAD4AIAAwACkAIAB1AG4AZABlAGYAaQBuAGUAZAA7AAoAIAAgAH0ACgAgACAAaQBmACAAKABEAGEAdABlAC4AbgBvAHcAKAApACAAPgAgADAAKQAgAHUAbgBkAGUAZgBpAG4AZQBkADsACgB9ACAAYwBhAHQAYwBoACAAewAKACAAIAB1AG4AZABlAGYAaQBuAGUAZAA7AAoAfQA%3D

### Expected result

No lint/correctness/noUnreachable error, code is reachable (and always reached).

The code is pretty weird but it's the minimum repro I found from my real code.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
